### PR TITLE
V2: совместимость и оптимизация операций со строками 

### PR DESCRIPTION
**Pull request by Mr-Rm**
Странноватое поведение `СтрНайти` проверено на последнем выпущенном релизе 8.3.27.1936

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation and clearer localized error messages for string operations and template processing.
  * String search behavior adjusted: default occurrence now targets the first match and input bounds are more strictly validated.

* **New Features**
  * Centralized, consistent error handling for string-related operations to produce more actionable messages.

* **Tests**
  * Expanded tests for string search covering empty inputs, invalid parameters, and boundary cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/OneScript.StandardLibrary/StringOperations.cs [hunk 1]
```diff
@@ -138,24 +147,28 @@ public int StrCompare(string first, string second)
...
-        public int StrFind(string haystack, string needle, SearchDirection direction = SearchDirection.FromBegin, int startPos = 0, int occurance = 0)
+        public int StrFind(string haystack, string needle, SearchDirection direction = SearchDirection.FromBegin, int startPos = 0, int occurance = 1)
         {
+            if (needle.Length == 0)
+                return 1;
+
             int len = haystack.Length;
-            if (len == 0 || needle.Length == 0)
+            if (len == 0)
                 return 0;
```
[the remark this answers is not in the document]
Comment (Mr-Rm) [reply src/OneScript.StandardLibrary/StringOperations.cs]: Поскольку параметры метода имеют конкретный тип `string`, а  не `IValue`, то при вызове пропущенный параметр будет передан как `""`, а не как `null`. Но для защиты при возможном прямом (не из скрипта) вызове можно проверку добавить
